src/providers/Anonymous.ts

```diff
@@ -28,7 +28,13 @@ import { Value } from "convex/values";
 /**
  * The available options to an {@link Anonymous} provider for Convex Auth.
  */
-export interface AnonymousConfig<DataModel extends GenericDataModel> {
+export interface AnonymousConfig<
+  DataModel extends GenericDataModel,
+  UserDocument extends Record<string, Value> = DocumentByName<
+    DataModel,
+    "users"
+  >,
```

Comment:
I wonder if we should just have this be the same as other providers, vs. returning a specific type.. not sure if this is causing the type issues